UCLA Radio/UCLA Radio/extensions/DateComponentsFormatter+String.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  DateComponentsFormatter+String.swift
+//  UCLA Radio
+//
+//  Created by Nathan Smith on 11/2/16.
+//  Copyright © 2016 ChrisLaganiere. All rights reserved.
+//
+
+import Foundation
+
+extension DateComponentsFormatter {
+    
+    // Expect string of format of "1pm"
+    func getHourComponentFromString(_ s: String) -> Int? {
+
+        let stringHour = s.substring(to: s.index(before: s.index(s.endIndex, offsetBy: -1)))
```

Comment:
actually a regular expression is perfect here, are you familiar with that? That's what this function did:

```
 -    static func matches(for regex: String!, in text: String!) -> [String] {
 -        
 -        do {
 -            let regex = try NSRegularExpression(pattern: regex, options: [])
 -            let nsString = text as NSString
 -            let results = regex.matches(in: text, options:[], range: NSMakeRange(0, nsString.length))
 -            return results.map { nsString.substring(with: $0.range)}
 -        } catch let error as NSError {
 -            print("invalid regex: \(error.localizedDescription)")
 -            return []
 -        }
 -    }
```

which let you use it like

```
let timestampA = "1pm"
let matchesA = matches(for: "[0-9]*", in: timestampA)
// matchesA: ["1"]

let timestampB = "1pm 2pm 13pm"
let matchesB = matches(for: "[0-9]*", in: timestampB)
// matchesB: ["1", "2", "13"]
```